subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -405,9 +755,222 @@ int fp_storage_ak_save(const struct fp_account_key *account_key)
 			"order may change at reboot.");
 	}
 
+	if (IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND) && ak_overwritten) {
+		/* Account Key overwritten. Remove bonds related with overwritten Account Key. */
+		err = invalid_bonds_purge(true);
+		if (err) {
+			LOG_ERR("invalid_bonds_purge failed (err %d). Ignoring the error and "
+				"continuing. There might be some invalid bonds present in the "
+				"Fast Pair AK storage subsystem.", err);
+		}
+	}
+
 	return 0;
 }
 
+static int ak_id_get(uint8_t *id, const struct fp_account_key *account_key)
+{
+	for (size_t i = 0; i < account_key_count; i++) {
+		if (!memcmp(account_key_list[i].key, account_key->key, FP_ACCOUNT_KEY_LEN)) {
+			*id = ACCOUNT_KEY_METADATA_FIELD_GET(account_key_metadata[i], ID);
+			return 0;
+		}
+	}
+
+	return -ESRCH;
+}
+
+void fp_storage_ak_bond_bt_request_cb_register(const struct fp_storage_ak_bond_bt_request_cb *cb)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+
+	__ASSERT_NO_MSG(cb && cb->bond_remove && cb->is_addr_bonded);
+
+	bt_request_cb = cb;
+}
+
+int fp_storage_ak_bond_conn_create(const void *conn_ctx, const bt_addr_le_t *addr,
+				   const struct fp_account_key *account_key)
+{
+	if (!IS_ENABLED(CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND)) {
+		__ASSERT_NO_MSG(false);
+		return -ENOTSUP;
+	}
+
+	if (!is_enabled) {
+		return -EACCES;
+	}
+
+	int err;
+	uint8_t id;
+	struct fp_bond_info *bond;
+	struct fp_bond_info bond_rollback_copy = (struct fp_bond_info){0};
+
+	if (account_key) {
+		err = ak_id_get(&id, account_key);
+		if (err) {
+			return err;
+		}
+	} else {
+		/* The actual ID will be determined during Account Key write. */
+		id = FP_BOND_INFO_AK_ID_UNKNOWN;
+	}
+
+	bond = fp_bond_info_free_get();
+	if (!bond) {
+		return -ENOMEM;
+	}
+
+	bond->conn_ctx = conn_ctx;
+	bond->bonded = false;
+	bond->nv.account_key_metadata_id = id;
+	bond->nv.addr = *addr;
+
+	err = fp_bond_info_settings_save(bond, &bond_rollback_copy);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+void fp_storage_ak_bond_conn_confirm(const void *conn_ctx, const bt_addr_le_t *addr)
```

Comment:
Consider validating that `addr` from the function argument matches address stored for the `bond_confirmed`. Maybe we could even skip passing `addr` as an argument here (assuming that we can rely on validity of address stored for `bond_confirmed`)

Reply:
I'll add assert here.